Wrappers/Python/cil/optimisation/utilities/StepSizeMethods.py

```diff
@@ -209,26 +213,35 @@ def get_step_size(self, algorithm):
             self.store_grad=algorithm.gradient_update.copy()# We store the last gradient in order to calculate the BB step size 
             return self.initial
         
+        gradient_norm = algorithm.gradient_update.norm()
         #If the gradient is zero, gradient based algorithms will not update and te step size calculation will divide by zero so we stop iterations. 
-        if algorithm.gradient_update.norm()<1e-8:
+        if gradient_norm < 1e-8:
             raise StopIteration
 
-        if self.mode=='long' or (self.mode =='alternate' and algorithm.iteration%2 ==0):
-            ret = (( algorithm.x-self.store_x).dot(algorithm.x-self.store_x))/ (( algorithm.x-self.store_x).dot(algorithm.gradient_update-self.store_grad))
-        elif self.mode=='short' or (self.mode =='alternate' and algorithm.iteration%2 ==1):
-            ret = (( algorithm.x-self.store_x).dot(algorithm.gradient_update-self.store_grad))/ (( algorithm.gradient_update-self.store_grad).dot(algorithm.gradient_update-self.store_grad))
+        algorithm.x.subtract(self.store_x, out=self.store_x) 
+        algorithm.gradient_update.subtract(self.store_grad, out=self.store_grad)
+        if self.current_mode == 'long':
+            ret = (self.store_x.dot(self.store_x))/ (self.store_x.dot(self.store_grad))
+        elif self.current_mode == 'short':
+            ret = (self.store_x.dot(self.store_grad))/ (self.store_grad.dot(self.store_grad))
         else:
             raise ValueError('Mode should be chosen from "long", "short" or "alternate". ')
```

Comment:
if you add this check in the init you can simplify this to an if/else

Reply:
I think done